src/Observables.jl

```diff
@@ -18,20 +18,55 @@ end
 """
 Like a `Ref` but updates can be watched by adding a handler using `on`.
 """
-mutable struct Observable{T} <: AbstractObservable{T}
-    id::String
-    val::T
+struct Observable{T} <: AbstractObservable{T}
+    id::UInt64
+    val::RefValue{T}
     listeners::Vector
 end
-Observable{T}(val) where {T} = Observable{T}(newid(), val, Any[])
-Observable(val::T) where {T} = Observable{T}(val)
+
+const observable_names = Dict{UInt64, String}()
+
+
+"""
+    rename!(x::AbstractObservable, name)
+
+Renames an observable
+"""
+rename!(x::AbstractObservable, name) = observable_names[obsid(x)] = string(name)
+
+"""
+    name!(x::AbstractObservable, name)
+
+Optional naming infrastructure for observables.
+name!(observable, "the name") will associate `observable` with the `the name`,
+and can be retrieved via `name(observable)`. If an observable already has a name,
+nothing will be done! Call `rename!(x::AbstractObservable, name)`, if you want to
+change the name.
+"""
+function name!(x::AbstractObservable, name)
+    id = obsid(x)
+    if !haskey(observable_names, id)
+        observable_names[id] = string(name)
+    end
+end
+
+"""
+    name(x::Observable, default = "noname")
+if observable `x` is named, returns the name and otherwise returns `default`.
+"""
+function name(x::Observable, default = "noname")
+    get(observable_names, obsid(x), default)
+end
+
+Observable{T}(val) where T = Observable{T}(newid(), RefValue{T}(val), Any[])
+Observable(val::T) where T = Observable{T}(val)
 
 observe(x::Observable) = x
 
-let count=0
+let count = UInt64(0)
     global newid
-    function newid(prefix="ob_")
-        string(prefix, lpad(count += 1, 2, "0"))
+    function newid()
+        count += 1
```

Comment:
This seems *much* better than using strings, I can't see any downside to this. Good move.